extensions/wikia/Email/Controller/EmailConfirmationController.class.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+namespace Email\Controller;
+
+use Email\EmailController;
+use Email\Check;
+use Email\Tracking\TrackingCategories;
+
+abstract class AbstractEmailConfirmationController extends EmailController {
+
+	const LAYOUT_CSS = EmailController::NO_ADDITIONAL_STYLES;
+
+	private $confirmUrl;
+
+	public function initEmail() {
+		$this->confirmUrl = $this->request->getVal( 'confirmUrl' );
+		$this->assertValidParams();
+	}
+
+	protected function assertValidParams() {
+		if ( empty( $this->confirmUrl ) ) {
+			throw new Check( "Must pass value for confirmUrl" );
+		}
+	}
+
+	/**
+	 * @template emailConfirmation
+	 */
+	public function body() {
+		$this->response->setData( [
+			'salutation' => $this->getSalutation(),
+			'summary' => $this->getSummary(),
+			'buttonLink' => $this->confirmUrl,
+			'buttonText' => $this->getMessage( 'emailext-emailconfirmation-button-text' ),
+			'contentFooterMessages' => $this->getContentFooterMessages()
+		] );
+	}
+
+	abstract protected  function getSummary();
+
+	protected function getContentFooterMessages() {
+		$commonFooterMessage = $this->getCommonFooterMessages();
+		$emailSpecificFooterMessages = $this->getEmailSpecificFooterMessages();
+		return array_merge( $emailSpecificFooterMessages, $commonFooterMessage );
+	}
+
+	private function getCommonFooterMessages() {
+		return [
+			$this->getMessage( 'emailext-emailconfirmation-community-team' )->text()
+		];
+	}
+
+	abstract protected function getEmailSpecificFooterMessages();
+
+	protected static function getEmailSpecificFormFields() {
+			$form = [
+				'inputs' => [
+					[
+						'type' => 'hidden',
+						'name' => 'confirmUrl',
+						'value' => "#",
+					],
+				]
+			];
+
+			return $form;
+
+	}
+}
+
+class EmailConfirmationController extends AbstractEmailConfirmationController {
+
+	const TRACKING_CATEGORY = TrackingCategories::EMAIL_CONFIRMATION;
```

Comment:
Just calling out these TRACKING_CATEGORY constants. This is how we'll categorize emails in sendgrid going forward and came out of that P2 (SOC-887).